src/SAD/ForTheL/Base.hs

```diff
@@ -441,25 +443,35 @@ such :: FTL ()
 such = tokenOf' ["such", "so"]
 
 
+--just for now:
+
+showVar :: VariableName -> Text
+showVar (VarConstant nm) = nm
+showVar nm = toLazyText $ represent nm
 
-texBegin, texEnd :: Text -> FTL Text
-texBegin env = do
+-- | Parses '\begin{env}'. Takes a parser for parsing 'env'.
+texBegin :: FTL a -> FTL a
+texBegin envType = do
   token "\\begin"
   symbol "{"
-  token env
+  envType' <- envType
   symbol "}"
-  return env
+  return envType'
 
-texEnd env = do
+-- | Parses '\end{env}'. Takes a parser for parsing 'env'.
+texEnd :: FTL () -> FTL ()
+texEnd envType = do
   token "\\end"
   symbol "{"
-  token env
+  envType
   symbol "}"
-  return env
 
+envLabel :: FTL Text
+envLabel = do
+  symbol "["
+  label <- anyToken
```

Comment:
I think this should be `topIdentifier`, because it fulfills the same function as the label in `Axiom Label. ...` and that uses `topIdentifier`

Reply:
Fixed!